hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java

```diff
@@ -446,15 +450,30 @@ public Hedera(
                         tssBaseService,
                         new FreezeServiceImpl(),
                         scheduleServiceImpl,
-                        new TokenServiceImpl(),
+                        new TokenServiceImpl(() -> {
+                            requireNonNull(initState);
+                            final var consensusTime =
+                                    ((MerkleStateRoot<?>) requireNonNull(initState)).getLastConsensusTime();
+                            final var stakePeriodManager = new StakePeriodManager(configProvider, instantSource);
+                            stakePeriodManager.setCurrentStakePeriodFor(consensusTime);
+                            final var calculator = new StakeRewardCalculatorImpl(stakePeriodManager);
+                            final var infoStore =
+                                    new WritableStakingInfoStore(initState.getWritableStates(TokenServiceImpl.NAME));
+                            final var rewardsStore = new ReadableNetworkStakingRewardsStoreImpl(
+                                    initState.getWritableStates(TokenServiceImpl.NAME));
+                            return ((MerkleStateRoot<?>) requireNonNull(initState))
+                                    .getPendingRewards(account -> calculator.computePendingReward(
```

Comment:
Since all these are related to `TokenService` state, can we directly do this calculation inside the schema?